File: src/typescript/src/models/SeaConditions.ts
/**
 * Represents the Sea Conditions section of SVD.
 */
export type SeaConditions = {
    SeaDirectionMagnetic: number
    SeaDirectionTrue: number
    SwellDirectionMagnetic: number
    SwellDirectionTrue: number
    WindDirectionMagnetic: number
    WindDirectionTrue: number
    BadWeatherHours: number
    BadWeatherDistance: number
    SeaHeight: number
    SwellHeight: number
    Current: number
}

File: src/dotnet/src/Ceataec.StandardizedVesselDataset/Models/SeaConditions.cs
namespace Ceataec.StandardizedVesselDataset.Models;

/// <summary>
/// Represent the Sea Conditions section of SVD
/// </summary>
public class SeaConditions
{
    public double SeaDirectionMagnetic { get; init; }
    public double SeaDirectionTrue { get; init; }
    public double SwellDirectionMagnetic { get; init; }
    public double SwellDirectionTrue { get; init; }
    public double WindDirectionMagnetic { get; init; }
    public double WindDirectionTrue { get; init; }
    public double BadWeatherHours { get; init; }
    public double BadWeatherDistance { get; init; }
    public double SeaHeight { get; init; }
    public double SwellHeight { get; init; }
    public double Current { get; init; }
}